sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs

```diff
@@ -101,41 +102,68 @@ public ArmClient(TokenCredential credential, ArmClientOptions options)
             if (credential is null)
                 throw new ArgumentNullException(nameof(credential));
 
-            Credential = credential;
             BaseUri = baseUri ?? new Uri(DefaultUri);
             options ??= new ArmClientOptions();
             if (options.Diagnostics.IsTelemetryEnabled)
```

Comment:
somewhat unrelated. Here you are modifying customer-provided instances of options.

Reply:
This is how we get the dynamic user-agent based on which restoperations is making the call

Reply:
You can pass it as a parameter to HttpPipelineBuilder.Build same as you do with auth policy https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/ManagementPipelineBuilder.cs#L19